graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -154,7 +169,9 @@ object Pregel extends Logging {
       // count the iteration
       i += 1
     }
-    messages.unpersist(blocking = false)
+    messageCheckpointer.unpersistDataSet()
```

Comment:
I think the thing is we use messageCheckpointer.update to do the cache, to make a pair, we can use it to unpersist data. Please correct me if I understand wrong. 
I think it's fine to add this new method as there is already a public method to cache data in PersistQueue, we should provide a public method to clean the queue.